vmod/vmod_debug.c

```diff
@@ -442,6 +442,9 @@ event_load(VRT_CTX, struct vmod_priv *priv)
 	if (cache_param->nuke_limit == 42) {
 		VSB_cat(ctx->msg, "nuke_limit is not the answer.");
 		return (-1);
+	} else if (!strcmp(VCL_Name(ctx->vcl), "debug_load")) {
```

Comment:
we should try to get to a test case not requiring magic vcl names

Reply:
Taking another look at it, I think we can simplify this test case a lot. It boils down to validating the assumption of "active VCL is always warm", and we don't acutally need a worker to do that for us. A simple assert in vmod_debug's `event_warm()` should suffice:

~~~
assert(vcl_active == NULL || vcl_active->temp->is_warm);
~~~

It may be the case that this assert should be placed somewhere in cache_vcl.c, but I am not seeing any obvious places to put it.